tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -104,6 +200,44 @@ def _test_seq_snapshots(context):
         microvm.kill()
 
 
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_patch_drive_snapshot(network_config,
+                              bin_cloner_path):
+    """Test scenario: 5 full sequential snapshots."""
+    logger = logging.getLogger("snapshot_sequence")
+
+    artifacts = ArtifactCollection(_test_images_s3_bucket())
+    # Testing matrix:
+    # - Guest kernel: Linux 4.9/4.14
+    # - Rootfs: Ubuntu 18.04
+    # - Microvm: 2vCPU with 512 MB RAM
+    # TODO: Multiple microvm sizes must be tested in the async pipeline.
+    microvm_artifacts = ArtifactSet(artifacts.microvms(keyword="2vcpu_512mb"))
+    kernel_artifacts = ArtifactSet(artifacts.kernels(keyword="4.14"))
+    disk_artifacts = ArtifactSet(artifacts.disks(keyword="ubuntu"))
+
+    # Create a test context and add builder, logger, network.
+    test_context = TestContext()
+    test_context.custom = {
+        'builder': MicrovmBuilder(bin_cloner_path),
+        'network_config': network_config,
+        'logger': logger
+    }
+
+    # Create the test matrix.
+    test_matrix = TestMatrix(context=test_context,
```

Comment:
Looks like the test matrix produces only 1 run configuration. I think you can drop the TestMatrix and move _test_patch_drive_snapshot code here.

Reply:
synced offline and decided to keep it like this since `TestMatrix` is responsible for downloading artifacts.

We plan to add microvm 'recipes' that download relevant artifacts and use them to replace current fixtures in a future PR.